prometheus-to-sd/translator/metric_descriptor_cache_test.go

```diff
@@ -67,3 +71,98 @@ func TestDescriptorChanged(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateMetricDescriptors(t *testing.T) {
```

Comment:
Can it be changed to incorporate all data in the test case? I.e. instead of filling the cache with everything and then checking something for each test case, create cache and populate it on each test case

Also, currently it's hard to understand what are the different cases, could you please add descriptions?

Reply:
Done, now it's much more readable. Thank you.